framework/decode/custom_vulkan_ascii_consumer.h

```diff
@@ -103,20 +103,21 @@ inline std::string DataPointerDecoderToString(const uint64_t buffer)
     return str;
 }
 
-/// Convert pointers to arrays of void pointers to a string of the captured address.
-template <typename T, typename U>
-inline std::string DataPointerDecoderToString(PointerDecoder<T, U>* pDecoder)
+/// Pointer overload for when the value of interest is passed indirectly.
+inline std::string DataPointerDecoderToString(const uint64_t* pBuffer)
 {
-    GFXRECON_ASSERT(pDecoder != nullptr &&
-                    "This pointer can never be null, even if the pointer from the capture encoded within it is null.");
-    std::string str{ "\"" + util::PtrToString(pDecoder->GetAddress()) + "\"" };
+    GFXRECON_ASSERT(nullptr != pBuffer &&
+                    "This should only ever come through as a pointer through an unavoidable weaknesses in "
+                    "the code gen: it should never be null.");
+    std::string str{ "\"" + util::PtrToString(*pBuffer) + "\"" };
     return str;
 }
 
-/// Convert void pointers into a string representation of the address in the capture file.
-inline std::string DataPointerDecoderToString(PointerDecoder<uint8_t>* pDecoder)
```

Comment:
I really liked having this concrete function to drop a breakpoint in during debugging and know exactly what types I was dealing with but the template version deals with this and also the `uint32_t`s pointed to by spirv code pointers, so the sensible choices were to lose the template and have concrete versions for all types or just have the template function to cover all PointerDecoder<> instantiations. Less code won.